sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -427,23 +429,51 @@ public static SerializationFormat serializationFormatFromContentType(HttpHeaders
     }
 
     /**
-     * Appends a query parameter to the given host URL.
+     * Appends a query parameter to the given URL.
      *
-     * @param url the base URL to which the query parameter will be appended.
-     * @param key the name of the query parameter (e.g., "api-version", "name", "After").
-     * @param value the value of the query parameter
-     * @return the updated URL with the appended query parameter.
+     * @param host The base URL to which the query parameter will be appended.
+     * @param queryParams A map containing the query parameters and their values.
+     * @param delimiter The delimiter to use if the value is a list.
+     * @return The URL with the appended query parameter.
      */
-    public static String appendQueryParam(String url, String key, String value) {
-        if (value == null) {
-            return url;
+    public static String appendQueryParam(String host, Map<String, Object> queryParams, char delimiter) {
```

Comment:
Should we remove `delimiter` and use the default `,`? When we have a usecase to add `,` we can add another overload when we need another delimiter and also, we don't have a way to specify the delimiter in the annotation today.